hack/extractcrd/main.go

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright 2022 The cert-manager Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
```

Comment:
note: This tool is added to extract a named CRD from a single YAML list of CRDs. In bazel we [generate one helm chart per CRD](https://github.com/cert-manager/cert-manager/blob/10c5d72279ae3b23f670338045c51ab10e808aa5/deploy/crds/BUILD.bazel#L58-L69) for this purpose, whereas I wanted to have one way of generating all CRDs and then extract them as needed.